agent/handlers/agentapi/v1/taskprotection/handlers/handlers.go

```diff
@@ -0,0 +1,136 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	apitask "github.com/aws/amazon-ecs-agent/agent/api/task"
+	"github.com/aws/amazon-ecs-agent/agent/engine/dockerstate"
+	"github.com/aws/amazon-ecs-agent/agent/handlers/agentapi/v1/taskprotection/types"
+	"github.com/aws/amazon-ecs-agent/agent/handlers/utils"
+	v3 "github.com/aws/amazon-ecs-agent/agent/handlers/v3"
+	"github.com/cihub/seelog"
+)
+
+// Returns endpoint path for PutTaskProtection API
+func TaskProtectionPath() string {
+	return fmt.Sprintf(
+		"/api/v1/%s/task/protection",
+		utils.ConstructMuxVar(v3.V3EndpointIDMuxName, utils.AnythingButSlashRegEx))
+}
+
+// Task protection request received from customers pending validation
+type taskProtectionRequest struct {
```

Comment:
Q. should we make it uppercase if we want to make it TaskProtectionRequest.ProtectionType or TaskProtectionRequest.ProtectionTimeoutMinutes public accessible?

Reply:
I think the fields should be lowercase. I had originally defined this type in `types` package with fields exported and I think I forgot to hide the fields when I moved this type here. Good catch!